plrust/src/user_crate/mod.rs

```diff
@@ -248,6 +251,35 @@ pub(crate) fn oid_to_syn_type(type_oid: &PgOid, owned: bool) -> Result<syn::Type
         .map_err(|e| PlRustError::ParsingRustMapping(type_oid.value(), rust_type.to_string(), e))
 }
 
+/// Builds a `Command::new("cargo")` with necessary environment variables pre-configured
+pub(crate) fn cargo() -> Command {
+    let mut command = Command::new("cargo");
+
+    if let Some(path) = PLRUST_PATH_OVERRIDE.get() {
+        // we were configured with an explicit $PATH to use
+        command.env("PATH", path);
+    } else {
+        let path = std::env::var("PATH");
+
+        if path == Err(VarError::NotPresent) || path == Ok(String::default()) {
+            // the environment has no $PATH, so lets try and make a good one based on where
+            // we'd expect 'cargo' to be installed
+            if let Some(home_directory) = dirs::home_dir() {
+                command.env(
+                    "PATH",
+                    format!("{}/.cargo/bin:/usr/bin", home_directory.display()),
```

Comment:
This is wrong on windows (I don't *think* it's wrong anywhere else, but am not certain), since `dirs::home_dir()` isn't where cargo places files. The right way would be to use <https://crates.io/crates/home>.

Also, this could use https://doc.rust-lang.org/nightly/std/env/fn.join_paths.html, but that's not that important.

Reply:
oh, I didn't know about the `home` crate.  I'll switch over to that.

I don't particularly care about Windows support since we don't, but best to future proof the easy things we can, I suppose.